### PR TITLE
Agent Get Pending Message

### PR DESCRIPTION
**Pull request by Future-Outlier**
# TL;DR
Relay detailed messages to the Flyte console to keep users informed of "Pending" status.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [x] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## ScreenShots
The test will pass in `base_agent.py` and `plugins/flytekit-spark/tests/test_agent.py::test_databricks_agent`

image

## Complete description
Two months ago, I introduced the [Databricks agent service](https://github.com/flyteorg/flytekit/pull/1797).
Currently, job statuses progress from "pending" to "running" to "succeeded".
To enhance user experience, @pingsutw, @kumare3, and I propose enriching the "pending" status with additional information for Databricks API and future implementations.
This update will relay detailed messages to the Flyte console to keep users informed.
## Follow-up issue
https://github.com/flyteorg/flyte/pull/4115
https://github.com/flyteorg/flyte/pull/4106
https://github.com/flyteorg/flyte/issues/3936

### CODE REVIEW

File: plugins/flytekit-spark/flytekitplugins/spark/agent.py [hunk 1]
```diff
@@ -70,10 +70,14 @@ async def async_get(self, context: grpc.ServicerContext, resource_meta: bytes) -
...
-        if response.get("state") and response["state"].get("result_state"):
-            cur_state = convert_to_flyte_state(response["state"]["result_state"])
-
-        return GetTaskResponse(resource=Resource(state=cur_state))
+        message = ""
+        if response.get("state"):
+            if response["state"].get("result_state"):
+                cur_state = convert_to_flyte_state(response["state"]["result_state"])
+            if response["state"].get("state_message"):
+                message = response["state"]["state_message"]
```
Comment (pingsutw) [inline plugins/flytekit-spark/flytekitplugins/spark/agent.py:78]: nit: could we update it to
```suggestion
       state = response.get("state")
        if state:
            if state.get("result_state"):
                cur_state = convert_to_flyte_state(state["result_state"])
            if state.get("state_message"):
                message = state["state_message"]
```

**Review by pingsutw — APPROVED** [review]
